### PR TITLE
[FIX] mail: no crash on livechat close in mobile

### PR DESCRIPTION
**Pull request by alexkuhn**
Steps to reproduce:

- Create a chatbot
- Browse your website as a user in moble
- Start a conversation with your chatbot
- Go back to your website or close the window

There was a crash because some code in chat window service makes sense only in `/web`.

Task-3367338

### CODE REVIEW

File: addons/mail/static/src/core/web/chat_window_service.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        if (this.ui.isSmall && !this.store.discuss.isActive) {
+            // If we are in mobile and discuss is not open, it means the
+            // chat window was opened from the messaging menu. In that
+            // case it should be re-opened to simulate it was always
+            // there in the background.
+            document.querySelector(".o_menu_systray i[aria-label='Messages']").click();
+            // ensure messaging menu is opened before chat window is closed
+            await Promise.resolve();
+        }
+        await this._super(...arguments);
```
Comment (seb-odoo) [inline addons/mail/static/src/core/web/chat_window_service.js:18]: I think `_super` needs to be saved before `await`.

File: addons/mail/static/src/core/web/chat_window_service.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/* @odoo-module */
+
+import { ChatWindowService } from "@mail/core/common/chat_window_service";
+
+import { patch } from "@web/core/utils/patch";
+
+patch(ChatWindowService.prototype, "mail/core/web", {
+    async close(chatWindow, { escape = false } = {}) {
```
Comment (seb-odoo) [inline addons/mail/static/src/core/web/chat_window_service.js:8]: useless params if you use arguments

**Review by tsm-odoo — APPROVED** [review]
@robodoo r+

File: addons/mail/static/src/core/web/chat_window_service.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+/* @odoo-module */
+
+import { ChatWindowService } from "@mail/core/common/chat_window_service";
+
+import { patch } from "@web/core/utils/patch";
+
+patch(ChatWindowService.prototype, "mail/core/web", {
+    async close() {
+        const _super = this._super.bind(this); // limitation of class.js
```
Comment (seb-odoo) [inline addons/mail/static/src/core/web/chat_window_service.js:9]: not class.js but patch?

Comment (tsm-odoo) [reply addons/mail/static/src/core/web/chat_window_service.js:9]: there is an await later

Comment (seb-odoo) [reply addons/mail/static/src/core/web/chat_window_service.js:9]: class.js is the legacy odoo class system, we don't use that anymore

Comment (alexkuhn) [reply addons/mail/static/src/core/web/chat_window_service.js]: removed this misleading comment, and also took opportunity to add suffix `_patch` in new file

**Comment by alexkuhn** [comment]
@robodoo r+

**Pull request closed without merging**